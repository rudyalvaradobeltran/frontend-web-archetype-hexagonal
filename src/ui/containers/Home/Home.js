import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

// Actions
import { userActions } from '../../../application/actions/user';

import HomeTemplate from '../../components/templates/HomeTemplate/HomeTemplate';

const Home = () => {
  const dispatch = useDispatch();
  const user = useSelector((store) => store.entities.user);

  useEffect(() => {
    dispatch(userActions.fetchUsers());
  }, []);

  return <HomeTemplate users={user.users} />;
};

export default Home;
