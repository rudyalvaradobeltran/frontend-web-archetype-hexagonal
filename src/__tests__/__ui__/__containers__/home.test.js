import React from 'react';
import { create } from 'react-test-renderer';

// Mocks
import MockProvider from '../../../__mocks__/mockProvider';
// Components
import Home from '../../../ui/containers/Home/Home';

describe('HOME CONTAINER', () => {
  it('Should render snapshot without crashing', () => {
    const wrapper = create(
      <MockProvider>
        <Home />
      </MockProvider>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
