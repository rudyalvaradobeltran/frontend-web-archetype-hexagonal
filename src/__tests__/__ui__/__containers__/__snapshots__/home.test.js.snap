// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HOME CONTAINER Should render snapshot without crashing 1`] = `
<div
  className="sc-gsDJrp hjRpwD"
/>
`;
